bin/network-monitor/src/counter.rs

```diff
@@ -0,0 +1,429 @@
+        let word = storage_slot.value();
+        let value = word.as_elements().last().expect("a word is always 4 elements").as_int();
+
+        Ok(Some(value))
+    } else {
+        Ok(None)
+    }
+}
+
+async fn setup_counter_increment(
+    config: MonitorConfig,
+    rpc_client: &mut RpcClient,
+) -> Result<(
+    CounterIncrementDetails,
+    Account,
+    Account,
+    BlockHeader,
+    MonitorDataStore,
+    NoteScript,
+    SecretKey,
+)> {
+    let details = CounterIncrementDetails::default();
+    // Load accounts from files
+    let (wallet_account, secret_key) = match create_wallet_account() {
+        Ok(account) => account,
+        Err(e) => {
+            error!("Failed to create wallet account: {:?}", e);
+            return Err(anyhow::anyhow!("Failed to create wallet account"));
+        },
+    };
+
+    let counter_account = match load_counter_account(&config.counter_file) {
+        Ok(account) => account,
+        Err(e) => {
+            error!("Failed to load counter account: {:?}", e);
+            return Err(anyhow::anyhow!("Failed to load counter account"));
+        },
+    };
+
+    // Get the genesis block header
+    let block_header = get_genesis_block_header(rpc_client).await?;
+
+    // Create the increment procedure script and get the library
+    let (increment_script, library) = create_increment_script()?;
+
+    // Create unified data store for transaction execution
+    let mut data_store = MonitorDataStore::new(block_header.clone(), PartialBlockchain::default());
+    data_store.add_account(wallet_account.clone());
+    data_store.add_account(counter_account.clone());
+    data_store.insert_library(&library);
+
+    Ok((
+        details,
+        wallet_account,
+        counter_account,
+        block_header,
+        data_store,
+        increment_script,
+        secret_key,
+    ))
+}
+
+/// Run the counter increment task.
+///
+/// This function periodically creates network notes that target the counter account
+/// and execute the increment procedure.
+///
+/// # Arguments
+///
+/// * `config` - The monitor configuration containing file paths and intervals.
+/// * `tx` - The watch channel sender for status updates.
+///
+/// # Returns
+///
+/// This function runs indefinitely, only returning on error.
+#[instrument(target = COMPONENT, name = "run-counter-increment-task", skip_all, ret(level = "debug"))]
+pub async fn run_counter_increment_task(
+    config: MonitorConfig,
+    tx: watch::Sender<ServiceStatus>,
+) -> Result<()> {
+    // Create RPC client
+    let mut rpc_client = create_rpc_client(&config).await?;
+
+    let (
+        mut details,
+        mut wallet_account,
+        counter_account,
+        block_header,
+        mut data_store,
+        increment_script,
+        secret_key,
+    ) = setup_counter_increment(config.clone(), &mut rpc_client).await?;
+
+    let mut rng = ChaCha20Rng::from_os_rng();
+
+    loop {
```

Comment:
This is a really large loop and is a bit hard to follow. Could we break up the logic into smaller functions within this loop?

Reply:
I moved the different paths into functions